components/slider-carousel/src/ReuseSliderCarousel.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useSlider } from "@locoworks/reusejs-toolkit-react-hooks";
+import { twMerge } from "tailwind-merge";
+
+interface ReuseSliderCarouselInterface {
+  slideInterval: number;
+  slides: React.ReactNode[];
+  loop: boolean;
+  wrapperClasses?: string;
+  sliderContainerClasses?: string;
+  enableButtons?: boolean | false;
+  prefixButton?: React.ReactNode;
+  suffixButton?: React.ReactNode;
+  buttonClasses?: string;
+  animationStyle?: string;
+}
+
+const ReuseSliderCarousel = ({
+  slideInterval,
+  slides,
+  loop,
+  wrapperClasses,
+  sliderContainerClasses,
+  enableButtons,
+  prefixButton,
+  suffixButton,
+  buttonClasses,
+  animationStyle,
+}: ReuseSliderCarouselInterface) => {
+  const {
+    currentSlide,
+    currentSlideIndex,
+    moveToNextSlide,
+    moveToPreviousSlide,
+    pauseSlider,
+    moveSlider,
+  } = useSlider({
+    slideInterval,
+    slides,
+    loop,
+  });
+
+  const parentRef = useRef<HTMLDivElement>(null);
```

Comment:
Can this ref be exposed and used outside the component?

Reply:
No, this ref cannot be exposed and used outside the component. The ref is intended for internal use within the component and is not designed to be exposed for external usage.